arroyo-state/src/lib.rs

```diff
@@ -416,6 +417,12 @@ impl<S: BackingStore> StateStore<S> {
     pub async fn load_compacted(&mut self, compaction: CompactionResult) {
         self.backend.load_compacted(compaction).await;
     }
+
+    pub async fn insert_committing_data(&mut self, table: char, committing_data: Vec<u8>) {
```

Comment:
Is committing data created fresh for each checkpoint? I.e., when I call this am I inserting it specifically in this checkpoint?

Following on from that:
* Does it make sense to include the epoch here so that everyone is aligned on what checkpoint we're talking about?
* In the future we will likely want to support aborting/timing out checkpoints; how would this interact with this system?

Reply:
I've added the epoch and a check on it. For aborting/timing out checkpoints we'll have to do some additional design. Currently the system is pretty optimistic that the checkpoint data will make it to state. When cancelling a checkpoint we'll have some subtasks that have already written that data, while others will have not received the barrier and should basically push all the data into the next checkpoint. I think it's doable, but a separate project.